File: client/src/lib/invoke/getNamesFromLogin.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { LoginNames } from "@bindings";

export function getNamesFromLogin(login: string): ResultAsync<LoginNames, string> {
    return ResultAsync.fromPromise(
        invoke("get_names_from_login", { login }),
        (error) => `Failed to get names from login: ${error}`
    );
}

File: client/src-tauri/src/commands/get_names_from_login.rs
use nexium::{login::Login, types::login_names::LoginNames};

#[tauri::command]
pub async fn get_names_from_login(login: String) -> Result<LoginNames, String> {
    // Utilise login::new(login) et login::get_names() pour obtenir le nom et le prénom
    Login::new(login)
        .map_err(|e| format!("Failed to create login: {}", e))?
        .get_names()
        .map_err(|e| format!("Failed to get names from login: {}", e))
}
